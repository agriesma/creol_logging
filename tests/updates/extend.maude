load creol-update .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> Configuration [ctor] .
  eq state =
    extend("BankAccount", (noInh), ("logger" |-> null),
      (< "transaction" : Method | Param: "sum", "s", Att: "sum" |-> null, "s"
         |-> null, "ccglab0" |-> null, "_" |-> null,
         Code: assign( "bal" ; "+" ( "bal" :: "sum" ) ) ;
         new( "logger" ; "Log" ; emp ) ; call( "ccglab0" ; "logger" ; "log" ;
         "s" ) ; get( "ccglab0" ; noVid ) ; return ( emp ) >),
      (skip), (c("BankAccount", 0)))
    
    < class("BankAccount", 0) : Class | Version: 0, Inh: noInh, Param: noVid,
      Att: "bal" |-> null,
      Mtds: < "balance" : Method | Param: noVid, Att: "_" |->
              null, "b" |-> null, Code: assign( "b" ; "bal" ) ;
              return ( "b" ) >,
            < "init" : Method | Param: noVid, Att: "_" |-> null,
              Code: assign( ( "bal" @ "BankAccount" ) ; int(0) ) ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null,
              Code: skip ; return ( emp ) >,
      Ocnt: 0 > .
endm

