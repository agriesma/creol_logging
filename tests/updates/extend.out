==========================================
rewrite in PROGRAM : state .
result Class: 
< class("BankAccount",1) : Class | Version: 1, Inh: noInh, Param: noVid, Att:
    "bal" |-> null, "logger" |-> null, Mtds:
  < ".update" : Method | Param: noVid, Att: ".version" |-> int(0), Code: skip
    >, < "balance" : Method | Param: noVid, Att: "_" |-> null, "b" |-> null,
    Code: assign("b" ; "bal") ; return("b") >, < "init" : Method | Param:
    noVid, Att: "_" |-> null, Code: assign("bal" @ "BankAccount" ; int(0)) ;
    return(emp) >, < "run" : Method | Param: noVid, Att: "_" |-> null, Code:
    skip ; return(emp) >, < "transaction" : Method | Param: "sum", "s", Att:
    "_" |-> null, "ccglab0" |-> null, "s" |-> null, "sum" |-> null, Code:
    assign("bal" ; "+"("bal" :: "sum")) ; new("logger" ; "Log" ; emp) ; call(
    "ccglab0" ; "logger" ; "log" ; "s") ; get("ccglab0" ; noVid) ; return(emp)
    >,
  Ocnt: 0 >

Bye.
